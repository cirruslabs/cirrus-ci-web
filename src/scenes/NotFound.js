import React from 'react';
import Icon from "@material-ui/core/Icon";

const NotFound = (props) => (
  <div style={{width: "100%", height: "100%"}}>
    <div className="row justify-content-between align-items-center"
         style={{width: "100%", height: "100%", fontSize: "32px"}}>
      <div className="col text-center">
        <Icon style={{fontSize: "96px"}} className="center-block">sentiment_very_dissatisfied</Icon>
        {
          props.messageComponent ? props.messageComponent :
            <div><p>{props.message || "Page not found! "}</p><a href="https://cirrus-ci.com" title="Homepage">Go home?</a></div>
        }
      </div>
    </div>
  </div>
);

export default NotFound;
