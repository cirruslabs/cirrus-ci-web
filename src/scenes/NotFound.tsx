import React, { useEffect } from 'react';
import Icon from '@mui/material/Icon';
import { Box, Link, Typography } from '@mui/material';

export default function NotFound(props) {
  useEffect(() => {
    document.title = 'Page Not Found - Cirrus CI';
  }, []);
  return (
    <Box sx={{ width: '100%', height: '100%', display: 'flex', justifyContent: 'center' }}>
      <Box
        sx={{
          flexDirection: 'column',
          alignItems: 'center',
          display: 'inline-flex',
        }}
      >
        <Icon style={{ fontSize: '96px' }}>sentiment_very_dissatisfied</Icon>
        {props.messageComponent ? (
          props.messageComponent
        ) : (
          <>
            <Typography variant="h2">{props.message || 'Page not found! '}</Typography>
            <Link href="https://cirrus-ci.com" color="inherit" variant="h3" title="Homepage">
              Go home?
            </Link>
          </>
        )}
      </Box>
    </Box>
  );
}
