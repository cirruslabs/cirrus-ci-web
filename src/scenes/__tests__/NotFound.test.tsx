import React from 'react';
import renderer from 'react-test-renderer';

import NotFound from 'scenes/NotFound';

it('renders correctly', () => {
  const tree = renderer.create(<NotFound />).toJSON();
  expect(tree).toMatchSnapshot();
});
