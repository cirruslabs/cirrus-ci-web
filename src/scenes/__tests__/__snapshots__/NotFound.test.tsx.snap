// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBox-root css-1aegtqj"
>
  <div
    className="MuiBox-root css-fhxkad"
  >
    <span
      aria-hidden={true}
      className="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
      style={
        Object {
          "fontSize": "96px",
        }
      }
    >
      sentiment_very_dissatisfied
    </span>
    <h2
      className="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
    >
      Page not found! 
    </h2>
    <a
      className="MuiTypography-root MuiTypography-h3 MuiLink-root MuiLink-underlineAlways css-1mgvoth-MuiTypography-root-MuiLink-root"
      href="https://cirrus-ci.com"
      onBlur={[Function]}
      onFocus={[Function]}
      title="Homepage"
    >
      Go home?
    </a>
  </div>
</div>
`;
